docs/en/aws-deploy-elastic-serverless-forwarder.asciidoc

```diff
@@ -230,6 +245,10 @@ For `elasticsearch` the following arguments are supported:
   * `args.api_key`:  API key of elasticsearch endpoint in the format `base64encode(api_key_id:api_key_secret)`. Mandatory when `args.username`  and `args.password` are not provided. Will be ignored if `args.username`/`args.password` are defined.
   * `args.es_datastream_name`: Name of data stream or index where logs should be forwarded to. Lambda supports automatic routing of various {aws} service logs to the corresponding data streams for further processing and storage in the {es} cluster. It supports automatic routing of `aws.cloudtrail`, `aws.cloudwatch_logs`, `aws.elb_logs`, `aws.firewall_logs`, `aws.vpcflow`, and `aws.waf` logs. For other log types, if using data streams, you can optionally set its value in the configuration file according to the naming convention for data streams and available integrations. If the `es_datastream_name` is not specified and it cannot be matched with any of the above {aws} services, then the value will be set to `logs-generic-default`. In versions **v0.29.1** and below, this configuration parameter was named `es_index_or_datastream_name`. Rename the configuration parameter to `es_datastream_name` in your `config.yaml` file on the S3 bucket to continue using it in the future version. The older name `es_index_or_datastream_name` is deprecated as of version **v0.30.0**. The related backward compatibility code is removed from version **v1.0.0**.
   * `args.es_dead_letter_index`: Name of data stream or index where logs should be redirected to, in case indexing to `args.es_datastream_name` returned an error.
+  * `es_dead_letter_forward_errors`: List of errors that should be forwarded to the dead letter index. The default value is an empty list (forward all errors). If the list is not empty, only the errors in the list will be forwarded to the dead letter index.
+    In general, you can use values from the `error.type` field in the Elasticseach API response. Here is a few examples of errors that can be forwarded to the dead letter index:
```

Comment:
> In general, you can use values from the `error.type`

What does this mean exactly? I see the condition to not be sent is:

```python
(  # no http status: connection error
                self._es_dead_letter_forward_errors
                and action["error"]["type"] not in self._es_dead_letter_forward_errors
            ):
```

I don't really understand the first part of the condition. It means you have to write the whole error to exclude it? Is there another way to exclude an error other than its `error.type`? @zmoog

Reply:
Sorry, I had the feeling this part wasn't clear enough.

> In general, you can use values from the `error.type`

The `error.type` in this sentence refers to the Elasticsearch API response.

For example, here's the Elasticsearch API response for 

```json
{
  "error": {
    "root_cause": [
      {
        "type": "fail_processor_exception",
        "reason": "Fail message"
      }
    ],
    "type": "fail_processor_exception",
    "reason": "Fail message"
  },
  "status": 500
}
```

`_parse_error()` picks `error.type` from the ES API response, so the shipper can later compare it with the values in the `_es_dead_letter_forward_errors` list.

> I don't really understand the first part of the condition. It means you have to write the whole error to exclude it? 

I don't get what you mean with "you have to write the whole error to exclude it". 

> Is there another way to exclude an error other than its error.type?

Users reported problems with the DLI forwarding all errors to ES. 

This PR focuses on adding essential filtering capability:

- do not forward connection-related errors (we assumed errors without `http.response.status_code` are related o connection failures).
- filter out errors based on their `error.type`

There are probably more ways to filter the errors based on other criteria. 

However, the Failure Store (FS) will ship in one upcoming ES release. So, it's better to give ESF what it needs today without re-implementing Failure Store features.

Reply:
@constanca-m I tried to clarify the `es_dead_letter_forward_errors` option docs: does it flow better?

Reply:
I will never ignore the feeling that a sentence isn't working again.

Thanks for raising the issue.

Reply:
@zmoog Can we add the clarification that `error.type` is the one returned by `bulk()` api? AFAIK, there's no exhaustive list of error types but that should be a good starting point